torch/cuda/comm.py

```diff
@@ -119,13 +131,20 @@ def reduce_add_coalesced(inputs, destination=None, buffer_size=10485760):
         A tuple of tensors containing an elementwise sum of each group of
         inputs, placed on the ``destination`` device.
     """
+    dense_tensors = []  # shape (num_tensors, num_gpus)
     output = []
-    itrs = [_take_tensors(tensors, buffer_size) for tensors in inputs]
+    for tensor_at_gpus in zip(*inputs):
+        if tensor_at_gpus[0].is_sparse:
```

Comment:
This is a bit worrying for me. If, for some reason, GPU 0 will produce a dense result, but GPU 1 will give you sparse grad, you will take the dense code path, and weird things will start happening